processor/lsmintervalprocessor/processor_test.go

```diff
@@ -352,19 +357,35 @@ func benchmarkAggregation(b *testing.B, ottlStatements []string) {
 			md, err := golden.ReadMetrics(filepath.Join(dir, "input.yaml"))
 			require.NoError(b, err)
 			md.MarkReadOnly()
-			b.ResetTimer()
-
-			err = mgp.Start(context.Background(), componenttest.NewNopHost())
-			require.NoError(b, err)
-			for i := 0; i < b.N; i++ {
+			benchFunc := func() {
+				b.StopTimer()
+				// Copy the metric as the ConsumeMetric call is mutating
 				mdCopy := pmetric.NewMetrics()
 				md.CopyTo(mdCopy)
 				// Overwrites the asdf attribute in metrics such that it becomes high cardinality
-				mdCopy.ResourceMetrics().At(0).Resource().Attributes().PutStr("asdf", fmt.Sprintf("%d", i))
+				mdCopy.ResourceMetrics().At(0).Resource().Attributes().PutStr("asdf", fmt.Sprintf("rand_%d", rand.Int()))
```

Comment:
Should each goroutine get its own `*rand.Rand`? The global one is synchronised, so this may artificially limit parallelism.

Alternatively, perhaps generate a random "ID" for each goroutine and pass that in, and add to the iteration to get a unique resource ID.